cluster_configs/barnacle/barnacle_status.py

```diff
@@ -0,0 +1,61 @@
+#!/usr/bin/env python
+
+import subprocess
+import sys
+import time
+import re
+from io import StringIO
+
+def parse_qstat(lines):
+    status = {}
+
+    for line in lines:
+        if line.startswith(' '):
+            entry = line.rstrip()
+            p = re.compile('^ +(.+?) = (.+)')
```

Comment:
I think this line should go above the for loop, so as to increase performance. A compiled regex can be reused over and over.

Reply:
Cool, did not know that!

Reply:
did that change happen?

Reply:
Ooops missed that one.